cmake/kokkos_arch.cmake

```diff
@@ -414,6 +433,8 @@ FUNCTION(CHECK_AMDGPU_ARCH ARCH FLAG)
       IF(KOKKOS_ENABLE_HIP_RELOCATABLE_DEVICE_CODE)
         GLOBAL_APPEND(KOKKOS_LINK_OPTIONS "${AMDGPU_ARCH_FLAG}=${FLAG}")
       ENDIF()
+      #GLOBAL_APPEND(KOKKOS_LINK_OPTIONS -lamdhip64 -lhsa-runtime64 -L${LIB_AMD_HIP_DIR})
+      GLOBAL_APPEND(KOKKOS_LINK_OPTIONS ${LIB_HSA_RUNTIME} ${LIB_AMD_HIP})
```

Comment:
Continuing with the find module approach, Kokkos is structured around defining imported or interface targets (e.g. Kokkos::ROCM) that would have an `INTERFACE_LINK_LIBRARIES=amdhip64 hsa-runtime64` and you would have a `target_link_libraries(kokkoscore PUBLIC Kokkos::ROCM)`